main.go

```diff
@@ -144,11 +144,12 @@ func main() {
 	setupLog.Info("setting up webhook server")
 	hookServer := mgr.GetWebhookServer()
 
+	decoder := admission.NewDecoder(mgr.GetScheme())
 	setupLog.Info("registering Subscription Channel validating webhook endpoint")
 	hookServer.Register("/validate-subscription", &webhook.Admission{
 		Handler: &admwebhook.SubscriptionAdmission{
 			Client:  mgr.GetClient(),
-			Decoder: admission.NewDecoder(mgr.GetScheme()),
+			Decoder: &decoder,
```

Comment:
Why this change? what was wrong with the old code? Even after the change the decoder is used only once

Reply:
In [CR v0.18](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission@v0.18.0#NewDecoder), `NewDecoder` returns argument of type `Decoder`, while what is required here is what used to be returned from [CR v0.17](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission@v0.17.2#NewDecoder) -- a pointer to one.

Reply:
A pointer to an interface does not make sense in almost all cases. 
When moving to an interface usually all places accepting a pointer to a struct will now accept an interface instead. 
I have a feeling that we have updated some libs but not all of the necessary libs, as part of the framework upgrade. 

Have we followed the upgrade notes? Sometimes there are manual changes that need to be applied

Reply:
Rebase allowed reverting this change